plugin/src/main/java/org/opensearch/ml/action/models/UpdateModelTransportAction.java

```diff
@@ -0,0 +1,280 @@
+                                    actionListener.onFailure(exception);
+                                }));
+                        } else {
+                            actionListener
+                                .onFailure(
+                                    new MLValidationException(
+                                        "User doesn't have privilege to perform this operation on this function category: "
+                                            + mlModel.getAlgorithm().toString()
+                                    )
+                                );
+                        }
+                    } catch (Exception e) {
+                        log.error("Failed to update ML model for model ID {}. Details {}:", modelId, e);
+                        actionListener.onFailure(e);
+                    }
+                } else {
+                    actionListener
+                        .onFailure(new IllegalArgumentException("Failed to find model to update with the provided model id: " + modelId));
+                }
+            }, e -> actionListener.onFailure(new MLResourceNotFoundException("Fail to find model"))));
+        } catch (Exception e) {
+            log.error("Failed to update ML model for " + modelId, e);
+            actionListener.onFailure(e);
+        }
+    }
+
+    private void updateModelWithRelinkStandAloneConnector(
+        String modelId,
+        String relinkModelGroupId,
+        String relinkConnectorId,
+        MLModel mlModel,
+        User user,
+        UpdateRequest updateRequest,
+        ActionListener<UpdateResponse> actionListener,
+        ThreadContext.StoredContext context
+    ) {
+        if (Strings.hasLength(mlModel.getConnectorId())) {
+            connectorAccessControlHelper
+                .validateConnectorAccess(client, relinkConnectorId, ActionListener.wrap(hasRelinkConnectorPermission -> {
+                    if (Boolean.TRUE.equals(hasRelinkConnectorPermission)) {
+                        updateModelWithOrWithoutRelinkModelGroup(modelId, relinkModelGroupId, user, updateRequest, actionListener, context);
+                    } else {
+                        actionListener
+                            .onFailure(
+                                new MLValidationException(
+                                    "You don't have permission to update the connector, connector id: " + relinkConnectorId
+                                )
+                            );
+                    }
+                }, exception -> {
+                    log.error("Permission denied: Unable to update the connector with ID {}. Details: {}", relinkConnectorId, exception);
+                    actionListener.onFailure(exception);
+                }));
+        } else {
+            actionListener
+                .onFailure(
+                    new IllegalArgumentException("This remote does not have a connector_id field, maybe it uses an internal connector.")
+                );
```

Comment:
Why don't we allow users to update model from inner connector to a standard connector?

Reply:
We will support update internal connector eventually after update connector API has been finalized, due to this part is a completely copy-paste to that one.

Reply:
So not supporting inner connector(for now) is by design.